controllers/storagecluster/external_resources.go

```diff
@@ -326,6 +339,108 @@ func (obj *ocsExternalResources) ensureDeleted(r *StorageClusterReconciler, inst
 	return reconcile.Result{}, nil
 }
 
+// claims should be created only once and should not be created/updated again if user deletes/update it.
+func (r *StorageClusterReconciler) createDefaultStorageClassClaims(instance *ocsv1.StorageCluster) error {
+
+	StorageClassClaimFile := &ocsv1alpha1.StorageClassClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      generateNameForCephFilesystemSC(instance),
+			Namespace: instance.Namespace,
+			Labels: map[string]string{
+				defaultStorageClassClaimLabel: "true",
+			},
+		},
+		Spec: ocsv1alpha1.StorageClassClaimSpec{
+			Type: "sharedfilesystem",
+		},
+	}
+
+	StorageClassClaimBlock := &ocsv1alpha1.StorageClassClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      generateNameForCephBlockPoolSC(instance),
+			Namespace: instance.Namespace,
+			Labels: map[string]string{
+				defaultStorageClassClaimLabel: "true",
+			},
+		},
+		Spec: ocsv1alpha1.StorageClassClaimSpec{
+			Type: "blockpool",
+		},
+	}
+
+	err := r.createAndOwnStorageClassClaim(instance, StorageClassClaimFile)
+	if err != nil {
+		return err
+	}
+
+	err = r.createAndOwnStorageClassClaim(instance, StorageClassClaimBlock)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (r *StorageClusterReconciler) createAndOwnStorageClassClaim(
```

Comment:
Can we refactor this function to take any number of claim objects? We can reduce redundant functions calls that way.

Reply:
done

Reply:
@iamniting Please undo as of comments from last reivew

Reply:
Which review comments? If there is a reason why my request doesn't make sense, I'd like to know about it and discuss.

Reply:
https://github.com/red-hat-storage/ocs-operator/pull/1607#discussion_r841260282

Reply:
@umangachapagain the change you requested is just another version of the loop that was there in the first place (see above comment). I don't really care if we have the ability to send multiple CRs to this function as long as we call it one at a time

Reply:
@umangachapagain 
I agree it is a little different, but my point remains. 
Imagen that for the block claim we need to introduce another new step, I think we can agree that both steps that need to apply to the first claim need to go together. 

Without the loop, the change is very straight straightforward, just add some other code after the first call to the function. But if we do it with the requested change, and the mentality of ODF engineers to not touch/refactor code that works, the result will be some new "magic" argument to the function and an `if` inside the function based on that "magic" value - I think we can agree that is no ideal!

If I was the one to write the code, I would not even introduce the function, I would embed the code in the main function, as that function does not have a meaningful logic that should really concern us.  

Every structure that we introduce into the code base is a limiting factor for future changes. 
As stated above, I would not mind that, if the culture was on this project was different.

Reply:
At the end of the day, this is very philosophical, and different experiences will result in different opinions. 
There is no actual bug or issue with both approaches, I would suggest letting the contribution decide as we already laied down here our opinions. 

@umangachapagain WDYT?

Reply:
I would like to keep as it is.